spec/services/versioned/publish_service_spec.rb

```diff
@@ -112,65 +109,67 @@
       end
 
       it "removes ones not used by the current edition" do
+        delete_request = stub_asset_manager_deletes_assets
+
         Versioned::PublishService.new(document)
                                  .publish(user: create(:user), with_review: true)
 
-        old_lead_image.reload
-        expect(old_lead_image.asset_manager_variants.map(&:state).uniq)
-          .to eq(%w[absent])
+        image_revision_to_remove.reload
+        expect(image_revision_to_remove.assets.map(&:state).uniq).to eq(%w[absent])
+        expect(delete_request).to have_been_requested.at_least_once
       end
 
       it "keeps images used by the current edition" do
+        stub_asset_manager_deletes_assets
         Versioned::PublishService.new(document)
                                  .publish(user: create(:user), with_review: true)
 
-        other_image.reload
-        expect(old_lead_image.asset_manager_variants.map(&:state).uniq)
-          .to eq(%w[live])
+        image_revision_to_keep.reload
+        expect(image_revision_to_keep.assets.map(&:state).uniq).to eq(%w[live])
       end
     end
 
     context "when the current edition has an updated version of the live editions lead image" do
```

Comment:
Is this just another 'it' of the previous context? Also, it's still referring to 'lead images' :-|.

Reply:
I'll update the previous context to make it more distinct, that one is checking we just remove the right images and this one checks we supersede images.

Reply:
Thanks re: lead image!

Reply:
👍